packages/js/src/generators/library/library.ts

```diff
@@ -262,17 +264,17 @@ function createFiles(tree: Tree, options: NormalizedSchema, filesDir: string) {
     tmpl: '',
     offsetFromRoot: offsetFromRoot(options.projectRoot),
     rootTsConfigPath: getRelativePathToRootTsConfig(tree, options.projectRoot),
-    buildable: options.buildable === true,
+    buildable: options.bundler && options.bundler !== 'none',
     hasUnitTestRunner: options.unitTestRunner !== 'none',
   });
 
-  if (options.compiler === 'swc') {
+  if (options.bundler === 'swc' || options.bundler === 'rollup') {
     addSwcDependencies(tree);
-    addSwcConfig(
-      tree,
-      options.projectRoot,
-      options.bundler === 'rollup' ? 'es6' : 'commonjs'
-    );
+    if (options.bundler === 'swc') {
+      addSwcConfig(tree, options.projectRoot, 'commonjs');
+    } else {
+      addSwcConfig(tree, options.projectRoot, 'es6');
+    }
```

Comment:
nit: we can go with this?
```suggestion
    addSwcConfig(tree, options.projectRoot, options.bundler === 'swc' ? 'commonjs' : 'es6');
```